File: src/PyTorch_CUDA_extension/name_classification/lstm.py
import time
import torch
import _ext.lstm
import numpy as np

class lstm(torch.autograd.Function):
    def __init__(self):
        super(lstm, self).__init__()

    def forward(self, x_data, weight, bias, hiddenSize, miniBatch, seqLength, numLayers):
        # if h_data.is_cuda == True and x_data.is_cuda == True and c_data.is_cuda == True:

        # start_t = time.time()

        _ext.lstm.lstm_forward(x_data, weight, bias, hiddenSize, miniBatch, seqLength, numLayers)
        # else:
        #     raise NotImplementedError()

        # elapsed_time = time.time() - start_t
        # print("Function wrapper time:\t%f seconds"%(elapsed_time))

        # be forced to return a Tensor by autograd.Function
        numElements = miniBatch*hiddenSize
        return x_data[:int(numElements)]

File: src/PyTorch_CUDA_extension/image_classification/src/lstm.c
#include <TH.h>
#include <THC.h>
#include <THCGeneral.h>
#include <time.h>

#include "lstm_kernel.h"

extern THCState* state;

int lstm_forward(
    THFloatTensor* x_data,
    THFloatTensor* weight,
    THFloatTensor* bias,
    THIntTensor* hiddenSize, 
    THIntTensor* miniBatch, 
    THIntTensor* seqLength, 
    THIntTensor* numLayers) {

	// clock_t start,end;
	// float e_time;
	// start = clock();
	
    forward(state, x_data, weight, bias, 
            hiddenSize, miniBatch, seqLength, numLayers);

	// end = clock();
	// e_time = ((float)(end - start)) / CLOCKS_PER_SEC;
	// printf("C wrapper time:\t%f\n", e_time);

    return 1;
}
